go/libraries/doltcore/env/memory.go

```
@@ -0,0 +1,229 @@
+		}
+
+		wsRef = ws.Ref()
+	}
+
+	return m.DoltDB.UpdateWorkingSet(ctx, wsRef, ws.WithStagedRoot(newRoot), h, m.workingSetMeta())
+}
+
+func (m MemoryRepoState) UpdateWorkingRoot(ctx context.Context, newRoot *doltdb.RootValue) error {
+	var h hash.Hash
+	var wsRef ref.WorkingSetRef
+
+	ws, err := m.WorkingSet(ctx)
+	if err == doltdb.ErrWorkingSetNotFound {
+		// first time updating root
+		wsRef, err = ref.WorkingSetRefForHead(m.CWBHeadRef())
+		if err != nil {
+			return err
+		}
+		ws = doltdb.EmptyWorkingSet(wsRef).WithWorkingRoot(newRoot).WithStagedRoot(newRoot)
+	} else if err != nil {
+		return err
+	} else {
+		h, err = ws.HashOf()
+		if err != nil {
+			return err
+		}
+
+		wsRef = ws.Ref()
+	}
+
+	return m.DoltDB.UpdateWorkingSet(ctx, wsRef, ws.WithWorkingRoot(newRoot), h, m.workingSetMeta())
+}
+
+func (m MemoryRepoState) WorkingSet(ctx context.Context) (*doltdb.WorkingSet, error) {
+	workingSetRef, err := ref.WorkingSetRefForHead(m.CWBHeadRef())
+	if err != nil {
+		return nil, err
+	}
+
+	workingSet, err := m.DoltDB.ResolveWorkingSet(ctx, workingSetRef)
+	if err != nil {
+		return nil, err
+	}
+
+	return workingSet, nil
+}
+
+func (m MemoryRepoState) workingSetMeta() *doltdb.WorkingSetMeta {
+	return &doltdb.WorkingSetMeta{
+		Timestamp:   uint64(time.Now().Unix()),
+		Description: "updated from dolt environment",
+	}
+}
+
+func (m MemoryRepoState) SetCWBHeadRef(_ context.Context, r ref.MarshalableRef) (err error) {
+	m.Head = r.Ref
+	return
+}
+
+func (m MemoryRepoState) GetRemotes() (map[string]Remote, error) {
+	return make(map[string]Remote), nil
+}
+
+func (m MemoryRepoState) AddRemote(name string, url string, fetchSpecs []string, params map[string]string) error {
+	return fmt.Errorf("cannot insert a remote in a memory database")
+}
+
+func (m MemoryRepoState) GetBranches() (map[string]BranchConfig, error) {
+	return make(map[string]BranchConfig), nil
+}
+
+func (m MemoryRepoState) UpdateBranch(name string, new BranchConfig) error {
+	return nil
```

Should throw an error